### PR TITLE
[pip][design] PIP-279: Reformat property in generateResponseWithEntry

### PR DESCRIPTION
**Pull request by StevenLuMT**
### Motivation

This is a PIP to reformat property in generateResponseWithEntry. The PR contents have the motivation.

### Documentation
- [x] `doc-not-needed` 

### Matching PR in forked repository
PR in forked repository: #20481

### CODE REVIEW

File: pip/pip-279.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Motivation
+
+reformat property,for a http header name cannot contain the following prohibited characters: =,;: \t\r\n\v\f
+
+for example:
+{"city=shanghai":"tag"}
+when we run `bin/pulsar-admin topics get-message-by-id `, it will throw exception, the exception is:
+`Reason: java.util.concurrent.CompletionException: org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector$RetryException: Could not complete the operation. Number of retries has been exhausted. Failed reason: a header name cannot contain the following prohibited characters: =,;: \t\r\n\v\f: =`
+
+<img width="1288" alt="image" src="https://github.com/StevenLuMT/pulsar/assets/42990025/973d95b9-4ac2-4977-b160-162c4b53a613">
```
Comment (eolivelli) [inline pip/pip-279.md:10]: Please don't link images not hosted in this repo

**Review by Technoboy- — APPROVED** [review]

File: pip/pip-279.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+for example:
+{"city=shanghai":"tag"}
+when we run `bin/pulsar-admin topics get-message-by-id `, it will throw exception, the exception is:
+`Reason: java.util.concurrent.CompletionException: org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector$RetryException: Could not complete the operation. Number of retries has been exhausted. Failed reason: a header name cannot contain the following prohibited characters: =,;: \t\r\n\v\f: =`
+
+# High Level Design
+
+In master branch, 
+in an http request:getMessageById("/{tenant}/{namespace}/{topic}/ledger/{ledgerId}/entry/{entryId}"),
+replace `"X-Pulsar-PROPERTY-" + msgProperties.getKey()` with `"X-Pulsar-PROPERTY"`
```
Comment (gaoran10) [inline pip/pip-279.md:14]: Could we add an example for this change? Such as this (inaccurate)
```
// message properties
a = value-a
b = value-b

// old response
header: {
  "X-Pulsar-PROPERTY-a": "value-a",
  "X-Pulsar-PROPERTY-b": "value-b",
}

// new response
header = {
  "X-Pulsar-PROPERTY": "{'a': 'value-a', 'b': 'value-b'}"
}
```

[the remark this answers is not in the document]
Comment (StevenLuMT) [reply pip/pip-279.md:14]: Good suggestion, I'm going to add this demo today @gaoran10

**Review by gaoran10 — APPROVED** [review]
+1 (non-binding)

**Review by AnonHxy — APPROVED** [review]

**Review by AnonHxy — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
+1

**Pull request merged**